hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -736,6 +736,7 @@ void Vgen::emit(const decqmlock& i) {
   if (RuntimeOption::EvalJitArmLse) {
     a->Mov(rVixlScratch0, -1);
     a->stadd(rVixlScratch0, adr);
+    a->Msr(NZCV, vixl::xzr);
```

Comment:
I'm not sure if just forcing the status flags to zero is the correct thing to do. The flags have to be set based on the arithmetic operation here, which is a subtraction of 1. Maybe instead you can `cmp` the result of the operation to zero? ...actually that's going to be hard seeing as how that result was set atomically in memory.

You might try making the use of `stadd` here an optimization for use only when the status flags don't need to be def'd. Otherwise if there is a use of the status flags that require this def, then revert to the original implementation. You can use the existing undef SF pass for guidance.

I have a suspicion that the status flags are always checked after a `decqmlock{}` as the primary use for this vasm instruction is during profiling for PGO. In that case, it might not be a huge deal if we just always use the old approach. Sure atomics will be faster, but this profiling only happens during warmup. AND #7984 actually introduces an option to just use a simple `decqm{}` to estimate the counters thus bypassing the use of `decqmlock{}` altogether.